bundles/org.openhab.binding.elgatokeylight/src/main/java/org/openhab/binding/elgatokeylight/internal/ElgatoKeyLight.java

```diff
@@ -0,0 +1,230 @@
+    private static final String BRIGHTNESS = "brightness";
+    private static final String ON = "on";
+
+    private static final String DISPLAY_NAME = "displayName";
+    private static final String SERIAL_NUMBER = "serialNumber";
+    private static final String PRODUCT_NAME = "productName";
+
+    private final String ip;
+    private final JsonParser parser;
+    private final Gson gson;
+    private volatile JsonObject accessoryInfo;
+    private final int port;
+
+    public ElgatoKeyLight(final String ip, final int port) {
+        this.ip = ip;
+        this.port = port;
+        parser = new JsonParser();
+        gson = new GsonBuilder().create();
+    }
+
+    private JsonObject createLightStatus(final Boolean on, final Integer brightness, final Integer temperature) {
+        JsonObject desiredLightState = new JsonObject();
+        desiredLightState.add("numberOfLights", new JsonPrimitive(1));
+        JsonArray lights = new JsonArray(1);
+        desiredLightState.add("lights", lights);
+        JsonObject light = new JsonObject();
+        lights.add(light);
+        if (on != null) {
+            light.addProperty(ON, on ? 1 : 0);
+        }
+        if (brightness != null) {
+            light.addProperty(BRIGHTNESS, brightness);
+        }
+        if (temperature != null) {
+            light.addProperty(TEMPERATURE, temperature);
+        }
+        return desiredLightState;
+    }
+
+    private JsonObject fetchAccessoryInfo() throws IOException {
+        String accessoryInfoResponse = HttpUtil.executeUrl(HttpMethod.GET.asString(), getUrl("/elgato/accessory-info"),
+                30);
+        if (accessoryInfoResponse == null) {
+            return null;
+        }
+
+        JsonElement accessoryInfo = parser.parse(accessoryInfoResponse);
+        if (!accessoryInfo.isJsonObject()) {
+            return null;
+        }
+
+        return (JsonObject) accessoryInfo;
+    }
+
+    private JsonObject fetchLightStatus() throws IOException {
+        String lightStatusResponse = HttpUtil.executeUrl(HttpMethod.GET.asString(), getUrl("/elgato/lights"), 30);
+        if (lightStatusResponse == null) {
+            return null;
+        }
+
+        JsonElement lightStatus = parser.parse(lightStatusResponse);
+        if (!lightStatus.isJsonObject()) {
+            return null;
+        }
+
+        return (JsonObject) lightStatus;
+    }
+
+    private JsonObject getAccessoryInfo() throws IOException {
+        JsonObject accessoryInfo = this.accessoryInfo;
+        while (accessoryInfo == null) {
+            accessoryInfo = this.accessoryInfo = fetchAccessoryInfo();
+        }
+        return accessoryInfo;
+    }
+
+    public String getIp() {
+        return ip;
+    }
+
+    private String getUrl(final String path) {
+        return "http://" + ip + ":" + port + path;
+    }
+
+    private LightStatus paresLightStatus(final JsonObject lightStatus) {
+        if ((lightStatus != null) && lightStatus.has("lights")) {
```

Comment:
The brackets are not needed as the != operator is more dominant than &&.
```suggestion
        if (lightStatus != null && lightStatus.has("lights")) {
```